docs/source/developers/crossbow.rst

```diff
@@ -0,0 +1,284 @@
+
+   .. code:: bash
+
+      conda install -c conda-forge -y \
+          jinja2 \
+          pygit2 \
+          click \
+          ruamel.yaml \
+          setuptools_scm \
+          github3.py \
+          python-gnupg \
+          toolz \
+          jira
+
+   .. code:: bash
+
+      # pygit2 requires libgit2: http://www.pygit2.org/install.html
+      pip install \
+          jinja2 \
+          pygit2 \
+          click \
+          ruamel.yaml \
+          setuptools_scm \
+          github3.py \
+          python-gnupg \
+          toolz \
+          jira
+
+9. Try running it:
+
+   .. code:: bash
+
+      $ python crossbow.py --help
+
+Usage
+-----
+
+The script does the following:
+
+1. Detects the current repository, thus supports forks. The following
+   snippet will build kszucs’s fork instead of the upstream apache/arrow
+   repository.
+
+   .. code:: bash
+
+      $ git clone https://github.com/kszucs/arrow
+      $ git clone https://github.com/kszucs/crossbow
+
+      $ cd arrow/dev/tasks
+      $ python crossbow.py submit conda-win conda-linux conda-osx
+
+2. Gets the HEAD commit of the currently checked out branch and
+   generates the version number based on `setuptools_scm`_. So to build
+   a particular branch, just check out before running the script:
+
+   .. code:: bash
+
+      git checkout ARROW-<ticket number>
+      python dev/tasks/crossbow.py submit --dry-run conda-linux conda-osx
+
+   ..
+
+      Note that the arrow branch must be pushed beforehand, because the
+      script will clone the selected branch.
+
+3. Reads and renders the required build configurations with the
+   parameters substituted.
+
+4. Create a branch per task, prefixed with the job id. For example to
+   build conda recipes on linux it will create a new branch:
+   ``crossbow@build-<id>-conda-linux``.
+
+5. Pushes the modified branches to GitHub which triggers the builds. For
+   authentication it uses GitHub OAuth tokens described in the install
+   section.
+
+Query the build status
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: bash
+
+   python crossbow.py status <build id / branch name>
+
+Download the build artifacts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: bash
+
+   python crossbow.py artifacts <build id / branch name>
+
+Examples
+~~~~~~~~
+
+The script accepts a pattern as a first argument to narrow the build
+scope:
+
+Run multiple builds:
+
+.. code:: bash
+
+   $ python crossbow.py submit debian-stretch conda-linux-gcc-py37
+   Repository: https://github.com/kszucs/arrow@tasks
+   Commit SHA: 810a718836bb3a8cefc053055600bdcc440e6702
+   Version: 0.9.1.dev48+g810a7188.d20180414
+   Pushed branches:
+    - debian-stretch
+    - conda-linux-gcc-py37
+
+Just render without applying or committing the changes:
+
+.. code:: bash
+
+   $ python crossbow.py submit --dry-run task_name
+
+Run only ``conda`` package builds and a Linux one:
+
+.. code:: bash
+
+   $ python crossbow.py submit -g conda centos-7
```

Comment:
What's -g?

Reply:
Renamed to the long name of the option `--group`